build.go

```diff
@@ -18,9 +18,24 @@ import (
 )
 
 type BuildOptions struct {
-	Indent int
-	Tabs   bool
-	Header bool
+	Indent         int
+	Tabs           bool
+	Header         bool
+	ExternalBuilds []Builder // handle specific directives
+}
+
+// Builder is the interface implemented by types that can render a Directive
+// as it appears in NGINX configuration files.
+//
+// RegisterBuilder returns the names of the directives for which the builder can
+// build NGINX configuration.
+//
+// Build writes the strings that represent the Directive and it's Block to the
```

Comment:
As implemented, the directive name is written to the string builder first before `Build` is called. Do we want to make that detail here (that the implementation is to return just the arguments and block) or do we want to make the implementation responsible for writing the directive name as well?